tests/go.mod

```diff
@@ -0,0 +1,16 @@
+module github.com/sanity-io/litter/tests
+
+go 1.23.7
```

Comment:
Might as well use latest (1.24.0?) and also update the root `go.mod`?

Reply:
Updated in tests module, shouldn't update the main one unless necessary, libraries automatically get compiled with the version of their consumer, this is a _minimum_, and it doesn't use any >1.16 features so I don't see why you'd break it for no reason